app/controllers/miq_ae_class_controller.rb

```diff
@@ -1029,14 +1029,19 @@ def method_form_fields
 
     if %w[ansible_job_template ansible_workflow_template].include?(location)
       # ManageIQ::Providers::AnsibleTower::Provider.where('zone_id != ?', Zone.maintenance_zone.id)
-      list_of_managers = ManageIQ::Providers::ExternalAutomationManager
-                         .where(:enabled => true)
-                         .pluck(:id, :name)
-                         .map { |r| {:id => r[0], :name => r[1]} }
+      list_of_managers = []
+      ManageIQ::Providers::ExternalAutomationManager.where(:enabled => true).each do |provider|
+        list_of_managers.push({:id => provider.id, :name => provider.name})
+      end
 
       if method&.options[:ansible_template_id]
-        manager_id = ManageIQ::Providers::ExternalAutomationManager::ConfigurationScript
-                     .find_by(:id => method.options[:ansible_template_id])&.manager_id
+        manager_id = if method.location == "ansible_workflow_template"
+                       ManageIQ::Providers::ExternalAutomationManager::ConfigurationWorkflow
+                         .find_by(:id => method.options[:ansible_template_id])&.manager_id
+                     else
+                       ManageIQ::Providers::ExternalAutomationManager::ConfigurationScript
+                         .find_by(:id => method.options[:ansible_template_id])&.manager_id
+                     end
```

Comment:
clearer:

```suggestion
        manager_class = method.location == "ansible_workflow_template" ? ManageIQ::Providers::ExternalAutomationManager::ConfigurationWorkflow : ManageIQ::Providers::ExternalAutomationManager::ConfigurationScript
        manager_id = manager_class.find_by(:id => method.options[:ansible_template_id])&.manager_id
```

Reply:
@agrare Thoughts here?  I really don't like the UI hardcoding ansible-specific stuff, but for now that might be the best way.

Reply:
I think the name "ansible_workflow_template" could be anything and doesn't have to be "ansible" it could be "configuration_workflow" or "configuration_script".

I do wonder if we need to make the distinction between the two, could we just let them pick any ConfigurationScript?  Is there a lot of different in how automate handles things if it is an ansible workflow vs ansible template?

Reply:
I guess that starts getting into supporting workflows and even Awx via automate methods?  I think we want to get there - not sure if it's as simple as just letting them pick it